Plugins~/Src/MeshSyncClientBlender/msblenModifiers.cpp

```diff
@@ -0,0 +1,327 @@
+
+		// Loop through modifier data and get the values:
+		for (auto property : blender::list_range((IDProperty*)nodeModifier->settings.properties->data.group.first)) {
+			if (property->name == attributeName) {
+				return IDP_Int(property);
+			}
+		}
+
+		return false;
+	}
+
+	void addModifierProperties(ModifierData* modifier, const Object* obj, ms::PropertyManager* propertyManager)
+	{
+		if (modifier->type != ModifierType::eModifierType_Nodes) {
+			return;
+		}
+
+		auto nodeModifier = (NodesModifierData*)modifier;
+		auto group = nodeModifier->node_group;
+
+		// Loop through modifier data and get the values:
+		for (auto property : blender::list_range((IDProperty*)nodeModifier->settings.properties->data.group.first)) {
+			if (strstr(property->name, "_use_attribute") || strstr(property->name, "_attribute_name")) {
+				continue;
+			}
+
+			if (doesPropertyUseAttribute(property->name, nodeModifier)) {
+				continue;
+			}
+
+			auto socket = getSocketForProperty(property, group);
+
+			if (socket != nullptr) {
+				auto propertyInfo = ms::PropertyInfo::create();
+
+				switch (property->type) {
+				case IDP_INT: {
+					if (socket->type == SOCK_BOOLEAN) {
+						propertyInfo->set(IDP_Int(property), 0, 1);
+					}
+					else {
+						auto defaultValue = (bNodeSocketValueInt*)socket->default_value;
+						propertyInfo->set(IDP_Int(property), defaultValue->min, defaultValue->max);
+					}
+					break;
+				}
+				case IDP_FLOAT: {
+					auto defaultValue = (bNodeSocketValueFloat*)socket->default_value;
+					propertyInfo->set(IDP_Float(property), defaultValue->min, defaultValue->max);
+					break;
+				}
+				case IDP_DOUBLE: {
+					auto defaultValue = (bNodeSocketValueFloat*)socket->default_value;
+					propertyInfo->set((float)IDP_Double(property), defaultValue->min, defaultValue->max);
+					break;
+				}
+				case IDP_ARRAY: {
+					auto defaultValue = (bNodeSocketValueVector*)socket->default_value;
+					switch (property->subtype) {
+					case IDP_INT: {
+						propertyInfo->set((int*)IDP_Array(property), defaultValue->min, defaultValue->max, property->len);
+						break;
+					}
+					case IDP_FLOAT: {
+						propertyInfo->set((float*)IDP_Array(property), defaultValue->min, defaultValue->max, property->len);
+						break;
+					}
+					}
+					break;
+				}
+				case IDP_STRING: {
+					auto uiData = (bNodeSocketValueString*)property->ui_data;
+					auto val = IDP_String(property);
+					propertyInfo->set(val, strlen(val));
+					break;
+				}
+				default:
+					continue;
+				}
+
+				propertyInfo->path = msblenUtils::get_path(obj);
+				propertyInfo->name = socket->name;
+				propertyInfo->modifierName = modifier->name;
+				propertyInfo->propertyName = property->name;
+				propertyInfo->sourceType = ms::PropertyInfo::SourceType::GEO_NODES;
+				propertyManager->add(propertyInfo);
+			}
+		}
+	}
+
+	void addCustomProperties(const Object* obj, ms::PropertyManager* propertyManager) {
+		if (obj->id.properties) {
```

Comment:
can this be inverted for an early exit if there are no properties?

Reply:
fixed